### PR TITLE
Replace -DLINUX/-DLINUX_ with __linux__

### PR DESCRIPTION
**Pull request by mattst88**

### CODE REVIEW

**Comment by dvrogozh** [comment]
@mattst88 : while that's a correct thing to do per-se, can you, please, provide your side motivation why you are doing this change? is there any functional or build time issue standing behind?

**Comment by mattst88** [comment]
> @mattst88 : while that's a correct thing to do per-se, can you, please, provide your side motivation why you are doing this change? is there any functional or build time issue standing behind?

I don't really have a side motivation; I just started helping to package this software in Gentoo Linux, glanced through the GitHub issues, and thought I'd see how easy or difficult it was to get contributions accepted from outside the company.

I guess if there's a side motivation, it's to work towards standardizing the build system to make it easier for packagers.

File: Tools/MediaDriverTools/GenDmyHex/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,4 @@ cmake_minimum_required (VERSION 2.8)
 project(IntelGenDmyHexTool) 
 add_compile_options(-std=c++11)
 
-add_definitions(-DLINUX_)
```
Comment (dvrogozh) [inline Tools/MediaDriverTools/GenDmyHex/CMakeLists.txt:25]: I think that all cmake level changes on Tools/MediaDriverTools/ level are fine. Don't remove them.

File: media_driver/cmake/linux/media_compile_flags_linux.cmake [hunk 2]
```diff
@@ -56,8 +56,6 @@ set(MEDIA_COMPILER_FLAGS_COMMON
     -m${ARCH}
 
     # Global defines
-    -DLINUX=1
```
Comment (dvrogozh) [inline media_driver/cmake/linux/media_compile_flags_linux.cmake:59]: Please, avoid dropping these defines.

File: media_softlet/cmake/linux/media_compile_flags_linux.cmake [hunk 3]
```diff
@@ -56,8 +56,6 @@ set(MEDIA_COMPILER_FLAGS_COMMON
     -m${ARCH}
 
     # Global defines
-    -DLINUX=1
```
Comment (dvrogozh) [inline media_softlet/cmake/linux/media_compile_flags_linux.cmake:59]: And here. Please, avoid dropping these defines.

**Review by dvrogozh — CHANGES_REQUESTED** [review]
@mattst88 : thank you, this aligns to my current understanding.

s/LINUX_/__linux__/ are all good. But changes where you remove self-defined macros are a little tricky because there is internal code which is not yet available in this public repo and it also uses these macros. And we have a sync in process to avoid code divergence. This sync eventually will be impacted. Thus, this change needs to be done in few stages:
1. Make all changes possible in this PR which won't impact internal flow
2. Someone internally need fix the internal code
3. After that cmake level change can be done 

Thus, may I ask you to remove some cmake level changes to make this PR easily mergeable? I'll mark them in inline comments.

**Comment by dvrogozh** [comment]
@jbeich : can you, please, evaluate how FreeBSD will be impacted by this change?

File: media_softlet/cmake/linux/media_compile_flags_linux.cmake (hunk 3 above)
Comment (mattst88) [reply media_softlet/cmake/linux/media_compile_flags_linux.cmake]: Thanks, done.

File: media_driver/cmake/linux/media_compile_flags_linux.cmake (hunk 2 above)
Comment (mattst88) [reply media_driver/cmake/linux/media_compile_flags_linux.cmake]: Thanks, done.

File: Tools/MediaDriverTools/GenDmyHex/CMakeLists.txt (hunk 1 above)
Comment (mattst88) [reply Tools/MediaDriverTools/GenDmyHex/CMakeLists.txt:25]: Thanks, I've kept all the CMake changes to Tools/MediaDriverTools.

**Comment by jbeich** [comment]
At #819 DragonFly, FreeBSD, NetBSD, OpenBSD and, tentively, Solarish (OpenIndiana) all currently use `-DLINUX` (explicitly defined) in order to avoid patch churn. With `__linux__` (peredefined macro) to support all those Unix-like platform one would need to replace `__linux__` with `__unix__` (predefined on Linux and BSDs but **not** macOS) or explicitly check for `__DragonFly__`, `__FreeBSD__`, `__NetBSD__`, `__OpenBSD__` and `__sun`.

**Comment by jbeich** [comment]
One negative side-effect of merging this PR would be rebase churn in #819. I usually don't rebase unless there's a patch conflict. The patches are used downstream on DragonFly, FreeBSD as cherry-picks and on OpenBSD as local copies... or were supposed to, I'm not sure if OpenBSD Ports has merged the media-driver package.
 https://github.com/freebsd/freebsd-ports/blob/main/multimedia/libva-intel-media-driver/Makefile#L7-L16

**Comment by mattst88** [comment]
I would be happy to rebase/redo this on top of your patch series, if Intel would accept yours. It's approaching its second birthday however...

**Comment by mattst88** [comment]
Another two weeks have passed.

Could someone explain what the actual process is for accepting PRs from Github? I see tags like 'verifying' which indicate to me that something is happening behind the scenes.

Is there someone upstream who is interested and able to merge external contributions?

**Comment by dvrogozh** [comment]
PR should pass review and get approval. Then someone on Intel side should trigger internal ci for the PR, if it will pass it will be merged. "verifying" label is to mark PRs which got picked for this internal ci.

In case of this PR we got https://github.com/intel/media-driver/pull/1296#issuecomment-982084427 which highlights that we can't take this patch as is without taking care of freebsd. I understand that we don't support freebsd right now w/o additional patches, but I don't want to make things worse than they are.

**Comment by mattst88** [comment]
Obsoleted by the first patch in https://github.com/intel/media-driver/pull/1785

**Pull request closed without merging**